rclc_parameter/include/rclc_parameter/rclc_parameter.h

```diff
@@ -365,6 +414,106 @@ rclc_parameter_get_double(
   const char * parameter_name,
   double * output);
 
+
+/**
+ * Add a description to a parameter (Unsuported on low memory mode).
+ * This description will be returned on describe parameters requests.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | No
+ *
+ * \param[in] parameter_server preallocated rclc_parameter_server_t
+ * \param[in] parameter_name name of the parameter
+ * \param[in] parameter_description description of the parameter
+ * \param[in] additional_constraints constraints description
+ * \param[in] read_only read only flag
+ * \return `RCL_RET_OK` if success
+ */
+RCLC_PARAMETER_PUBLIC
+rcl_ret_t
+rclc_add_parameter_description(
+  rclc_parameter_server_t * parameter_server,
+  const char * parameter_name, const char * parameter_description,
+  const char * additional_constraints);
```

Comment:
Why do we have the same API for setting the description and the constraints?
Is not possible to set constraints or descriptions independently?

Reply:
This method sets only the strings descriptions, while `rclc_add_parameter_constraints_integer/double` sets numeric constrains.

- `additional_constraints` may be unrelated to numeric values, as the user may use this to advertise its own callback rules.
- The description string values are unsupported on the low memory version, that is why I grouped this values on the same method:
https://github.com/ros2/rclc/blob/b600abd672c115aca3a8680f49b7e090ba9f4b2f/rclc_parameter/include/rclc_parameter/rclc_parameter.h#L419

Maybe we could divide this in `rclc_add_parameter_description` and `rclc_add_constraints_description`.

Reply:
Ok with that